src/components/postgresConnect.ts

```diff
@@ -594,13 +594,19 @@ class PostgresDB {
     }
   }
 
-  public updateTriggerQueueStatus = async (
-    payload: TriggerQueueUpdatePayload
-  ): Promise<boolean> => {
-    const text = 'UPDATE trigger_queue SET status = $1 WHERE id = $2'
-    // TODO: Dynamically select what is being updated
-    try {
-      await this.query({ text, values: Object.values(payload) })
+  public updateTriggerQueueStatus = async ({
+    id,
+    application_id,
+    status,
+  }: TriggerQueueUpdatePayload): Promise<boolean> => {
+    const text = `UPDATE trigger_queue SET status = $1${
+      application_id ? ', application_id = $3' : ''
+    } WHERE id = $2`
+    const values = [status, id]
+    if (application_id) values.push(application_id)
+
+    try {
+      await this.query({ text, values })
```

Comment:
I don't quite understand the mechanism of this code, but it is working well - I'll take some time later to properly wrap my head around it.

Reply:
This bit is a bit messy, but it's just building a conditional SQL query, because the `application_id` may or may not be present, so the query string has to change accordingly. You'll see other examples of this around such as when the `orgId` might be present or not.